File: app/src/test/java/com/vrem/wifianalyzer/wifi/graphutils/GraphAdapterTest.java
/*
 * WiFiAnalyzer
 * Copyright (C) 2019  VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>
 */

package com.vrem.wifianalyzer.wifi.graphutils;

import com.jjoe64.graphview.GraphView;
import com.vrem.wifianalyzer.MainContextHelper;
import com.vrem.wifianalyzer.wifi.model.WiFiData;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import java.util.Collections;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class GraphAdapterTest {
    @Mock
    private GraphViewNotifier graphViewNotifier;
    @Mock
    private GraphView graphView;
    @Mock
    private WiFiData wifiData;

    private GraphAdapter fixture;

    @Before
    public void setUp() {
        fixture = new GraphAdapter(Collections.singletonList(graphViewNotifier));
    }

    @After
    public void tearDown() {
        MainContextHelper.INSTANCE.restore();
    }

    @Test
    public void testUpdate() {
        // execute
        fixture.update(wifiData);
        // validate
        verify(graphViewNotifier).update(wifiData);
    }

    @Test
    public void testGetGraphViews() {
        // setup
        when(graphViewNotifier.getGraphView()).thenReturn(graphView);
        // execute
        List<GraphView> actual = fixture.getGraphViews();
        // validate
        assertEquals(1, actual.size());
        assertEquals(graphView, actual.get(0));
        verify(graphViewNotifier).getGraphView();
    }

}

File: app/src/test/kotlin/com/vrem/wifianalyzer/wifi/graphutils/GraphAdapterTest.kt
/*
 * WiFiAnalyzer
 * Copyright (C) 2015 - 2026 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>
 */
package com.vrem.wifianalyzer.wifi.graphutils

import android.view.View
import com.vrem.wifianalyzer.MainContextHelper
import com.vrem.wifianalyzer.wifi.model.WiFiData
import org.assertj.core.api.Assertions.assertThat
import org.junit.After
import org.junit.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify
import org.mockito.kotlin.verifyNoMoreInteractions
import org.mockito.kotlin.whenever

class GraphAdapterTest {
    private val graphNotifier: GraphNotifier = mock()
    private val view: View = mock()
    private val wiFiData: WiFiData = mock()
    private val fixture = GraphAdapter(listOf(graphNotifier))

    @After
    fun tearDown() {
        verifyNoMoreInteractions(graphNotifier)
        verifyNoMoreInteractions(view)
        verifyNoMoreInteractions(wiFiData)
        MainContextHelper.INSTANCE.restore()
    }

    @Test
    fun update() {
        // execute
        fixture.update(wiFiData)
        // validate
        verify(graphNotifier).update(wiFiData)
    }

    @Test
    fun destroy() {
        // Act
        fixture.destroy()
        // Assert
        verify(graphNotifier).destroy()
    }

    @Test
    fun graphs() {
        // setup
        whenever(graphNotifier.graph()).thenReturn(view)
        // execute
        val actual = fixture.graphs()
        // validate
        assertThat(actual).hasSize(1)
        assertThat(actual[0]).isEqualTo(view)
        verify(graphNotifier).graph()
    }
}
